Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -2834,30 +2834,46 @@ - (void)yapDatabaseModified:(NSNotification *)notification
     // We need to reload any modified interactions _before_ we call
     // reloadViewItems.
     BOOL hasDeletions = NO;
+    BOOL hasMalformedRowChange = NO;
     for (YapDatabaseViewRowChange *rowChange in rowChanges) {
         switch (rowChange.type) {
             case YapDatabaseViewChangeUpdate: {
                 YapCollectionKey *collectionKey = rowChange.collectionKey;
-                OWSAssert(collectionKey.key.length > 0);
                 if (collectionKey.key) {
                     ConversationViewItem *viewItem = self.viewItemCache[collectionKey.key];
                     [self reloadInteractionForViewItem:viewItem];
+                } else {
+                    hasMalformedRowChange = YES;
                 }
                 break;
             }
             case YapDatabaseViewChangeDelete: {
                 // Discard cached view items after deletes.
                 YapCollectionKey *collectionKey = rowChange.collectionKey;
-                OWSAssert(collectionKey.key.length > 0);
                 if (collectionKey.key) {
                     [self.viewItemCache removeObjectForKey:collectionKey.key];
+                } else {
+                    hasMalformedRowChange = YES;
                 }
                 hasDeletions = YES;
                 break;
             }
             default:
                 break;
         }
+        if (hasMalformedRowChange) {
+            break;
+        }
+    }
+
+    if (hasMalformedRowChange) {
+        // These errors seems to be very rare; they can only be reproduced
+        // using the more extreme actions in the debug UI.
```

Comment:
If you insert 100k messages into a conversation, YapDatabase's "group change" starts yielding bad data.  I'm not sure if this'll make us any more robust in production, but it can't hurt.